easycv/apis/export.py

```diff
@@ -145,31 +154,118 @@ def _export_cls(model, cfg, filename):
+            raise ValueError('`end2end` only support torch1.7.0 and later!')
+
+        batch_size = cfg.export.get('batch_size', 1)
+        img_scale = cfg.get('img_scale', (640, 640))
+        assert (
+            len(img_scale) == 2
+        ), 'Export YoloX predictor config contains img_scale must be (int, int) tuple!'
+
+        input = 255 * torch.rand((batch_size, 3) + img_scale)
+
+        model_export = End2endModelExportWrapper(
+            model,
+            input.to(device),
+            preprocess_fn=PreProcess(target_size=(640,
+                                                  640)) if end2end else None,
+            postprocess_fn=DetPostProcess(max_det=100, score_thresh=0.5)
+            if end2end else None,
+            trace_model=True,
+        )
 
-    meta = dict(config=json.dumps(config))
-    checkpoint = dict(
-        state_dict=model.state_dict(), meta=meta, author='EasyCV')
-    with io.open(filename, 'wb') as ofile:
-        torch.save(checkpoint, ofile)
+        model_export.eval().to(device)
+
+        # well trained model will generate reasonable result, otherwise, we should change model.test_conf=0.0 to avoid tensor in inference to be empty
+        # use trace is a litter bit faster than script. But it is not supported in an end2end model.
+        if end2end:
+            try:
+                yolox_trace = torch.jit.script(model_export)
+            except:
+                assert 'jit export error may occur when the model is not well trained'
+        else:
+            try:
+                yolox_trace = torch.jit.trace(model_export, input.to(device))
+            except:
+                assert 'jit export error may occur when the model is not well trained'
+
+        if getattr(cfg.export, 'export_blade', False):
+            blade_config = cfg.export.get('blade_config',
+                                          dict(enable_fp16=True))
+
+            from easycv.toolkit.blade import blade_env_assert, blade_optimize
+
+            if blade_env_assert() == True:
+
+                if end2end:
+                    input = 255 * torch.rand(img_scale + (3, ))
+
+                yolox_blade = blade_optimize(
+                    script_model=model,
+                    model=yolox_trace,
+                    inputs=(input.to(device), ),
+                    blade_config=blade_config)
+
+                with io.open(filename + '.blade', 'wb') as ofile:
+                    torch.jit.save(yolox_blade, ofile)
+                with io.open(filename + '.blade.config.json', 'w') as ofile:
+                    config = dict(
+                        export=cfg.export,
+                        test_pipeline=cfg.test_pipeline,
+                        classes=cfg.CLASSES)
+
+                    # meta = dict(config=json.dumps(config))
+                    json.dump(config, ofile)
+            else:
+                logging.warning('Export YoloX predictor with blade failed!')
```

Comment:
if export_blade=True，is it better to raise an error directly instead of logging a warning?